src/main/java/org/broadinstitute/hellbender/tools/exome/ReadCountCollectionUtils.java

```diff
@@ -561,8 +563,51 @@ public static ReadCountCollection removeColumnsWithExtremeMedianCounts(final Rea
         }
     }
 
-    private static long countZeroes(final double[] data) {
-        return DoubleStream.of(data).filter(d -> d == 0.0).count();
+    /**
+     * Remove columns with NaNs and infinities
+     *
+     * @param readCounts
+     * @param logger
+     * @return
+     */
+    public static ReadCountCollection removeColumnsWithBadValues(final ReadCountCollection readCounts,
+                                                                 final Logger logger) {
+        final List<String> columnNames = readCounts.columnNames();
+
+        // Determine kept and dropped column sets.
+        final Set<String> columnsToKeep = new LinkedHashSet<>(readCounts.columnNames().size());
+        final Set<String> columnsToDrop = new HashSet<>();
+        for (int i = 0; i < columnNames.size(); i++) {
+            if (Arrays.stream(readCounts.getColumn(i))
```

Comment:
I would just make this a nested for loop, where the inner loop scans the column and breaks once a bad value is found.